src/biogeophys/SnowSnicarMod.F90

```diff
@@ -338,6 +357,175 @@ subroutine SNICAR_RT (flg_snw_ice, bounds, num_nourbanc, filter_nourbanc,  &
+    integer :: ngmax = 8                          ! maxmimum gaussian integration index
+    ! constants used in algorithm
+    real(r8):: c0     = 0.0_r8
+    real(r8):: c1     = 1.0_r8
+    real(r8):: c3     = 3.0_r8
+    real(r8):: c4     = 4.0_r8
+    real(r8):: c6     = 6.0_r8
+    real(r8):: cp01   = 0.01_r8
+    real(r8):: cp5    = 0.5_r8
+    real(r8):: cp75   = 0.75_r8
+    real(r8):: c1p5   = 1.5_r8
+    real(r8):: trmin  = 0.001_r8
+    real(r8):: argmax = 10.0_r8                   ! maximum argument of exponential
+    ! cconstant and coefficients used for SZA parameterization
+    real(r8):: sza_a0 =  0.085730_r8
+    real(r8):: sza_a1 = -0.630883_r8
+    real(r8):: sza_a2 =  1.303723_r8
+    real(r8):: sza_b0 =  1.467291_r8
+    real(r8):: sza_b1 = -3.338043_r8
+    real(r8):: sza_b2 =  6.807489_r8
+    real(r8):: puny   =  1.0e-11_r8
+    real(r8):: mu_75  =  0.2588_r8                ! cosine of 75 degree
+    real(r8):: sza_c1                             ! coefficient, SZA parameteirzation
+    real(r8):: sza_c0                             ! coefficient, SZA parameterization
+    real(r8):: sza_factor                         ! factor used to adjust NIR direct albedo
+    real(r8):: flx_sza_adjust                     ! direct NIR flux adjustment from sza_factor
+    real(r8):: mu0                                ! incident solar zenith angle
+
+    !-----------------------------------------------------------------------
+    ! variables used for nonspherical snow grain treatment (He et al. 2017 J of Climate):
+    integer  :: sno_shp(-nlevsno+1:0)             ! Snow shape type: 1=sphere; 2=spheroid; 3=hexagonal plate; 4=koch snowflake
+                                                  ! currently only assuming same shapes for all snow layers
+    real(r8) :: sno_fs(-nlevsno+1:0)              ! Snow shape factor: ratio of nonspherical grain effective radii to that of equal-volume sphere
+                                                  ! only activated when snicar_snw_shape > 1 (i.e. nonspherical)
+                                                  ! 0=use recommended default value (He et al. 2017);
+                                                  ! others(0<sno_fs<1)= user-specified value
+    real(r8) :: sno_AR(-nlevsno+1:0)              ! Snow grain aspect ratio: ratio of grain width to length
+                                                  ! only activated when snicar_snw_shape > 1 (i.e. nonspherical)
+                                                  ! 0=use recommended default value (He et al. 2017);
+                                                  ! others(0.1<fs<20)= use user-specified value
+    ! Constants and parameters for aspherical ice particles    
+    ! asymmetry factor parameterization coefficients (6 bands) from Table 3 & Eqs. 6-7 in He et al. (2017)
+    real(r8) :: g_wvl(1:8)                        ! wavelength (um) division point
```

Comment:
In this section (and wherever else it applies) I recommend replacing the array dimensions with some explanation of what they are. E.g. why 8 vs. 7 vs. 6 vs. 16 vs. 17 etc.

Reply:
these numbers are based on the original parameterization development for the number of wavelength bands.

Reply:
Looking at these again:
Do the array dimensions sized 8 and 7 corresponding to `ngmax` and `ngmax - 1`?

Reply:
I resolved a very small number of these for which I felt confident. Others I left as hardwired numbers.